cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -1622,16 +1623,24 @@ public CloudTask runTask(String applicationName, CloudTask task) {
         Map<String, Object> urlVariables = new HashMap<>();
         urlVariables.put("applicationGuid", applicationId);
         Map<String, Object> request = new HashMap<>();
-        request.put("name", task.getName());
+        Assert.notNull(task.getCommand(), "Command must not be null");
         request.put("command", task.getCommand());
-        request.put("memory_in_mb", task.getMemory());
-        request.put("disk_in_mb", task.getDiskQuota());
+
+        addNonNullRequestParameter(request, "name", task.getName());
+        addNonNullRequestParameter(request, "memory_in_mb", task.getMemory());
+        addNonNullRequestParameter(request, "disk_in_mb", task.getDiskQuota());
 
         @SuppressWarnings("unchecked")
         Map<String, Object> resource = getRestTemplate().postForObject(getUrl(urlPath), request, Map.class, urlVariables);
         return resourceMapper.mapResource(resource, CloudTask.class);
     }
 
+    private void addNonNullRequestParameter(Map<String, Object> request, String requestParameterName, Object value) {
+        if (!Objects.isNull(value)) {
```

Comment:
Just use a standard null check. Objects.isNull is meant to be used as a predicate in streams, i.e. filter(Objects::isNull).